pkg/sql/sqlbase/rowwriter.go

```diff
@@ -518,33 +518,38 @@ func makeRowUpdaterWithoutCascader(
 		tableCols = append(tableCols, tableDesc.Columns...)
 	}
 
-	var deleteOnlyIndex map[int]struct{}
+	var deleteOnlyIndexes []IndexDescriptor
 	for _, m := range tableDesc.Mutations {
 		if index := m.GetIndex(); index != nil {
 			if needsUpdate(*index) {
-				indexes = append(indexes, *index)
-
 				switch m.State {
 				case DescriptorMutation_DELETE_ONLY:
-					if deleteOnlyIndex == nil {
+					if deleteOnlyIndexes == nil {
 						// Allocate at most once.
-						deleteOnlyIndex = make(map[int]struct{}, len(tableDesc.Mutations))
+						deleteOnlyIndexes = make([]IndexDescriptor, 0)
```

Comment:
deleteOnlyIndexes = make([]IndexDescriptor, 0, len(tableDesc.Mutations)) so it's preallocated optimally